store/tikv/mock-tikv/rpc.go

```diff
@@ -0,0 +1,319 @@
+	return &kvrpcpb.CmdPrewriteResponse{
+		Errors: extractKeyErrors(errors),
+	}
+}
+
+func (h *rpcHandler) onCommit(req *kvrpcpb.CmdCommitRequest) *kvrpcpb.CmdCommitResponse {
+	for _, k := range req.Keys {
+		if !h.keyInRegion(k) {
+			panic("onCommit: key not in region")
+		}
+	}
+	errors := h.mvccStore.Commit(req.Keys, req.GetStartVersion(), req.GetCommitVersion())
+	return &kvrpcpb.CmdCommitResponse{
+		Errors: []*kvrpcpb.KeyError{extractKeyError(errors)},
+	}
+}
+
+func (h *rpcHandler) onCleanup(req *kvrpcpb.CmdCleanupRequest) *kvrpcpb.CmdCleanupResponse {
+	if !h.keyInRegion(req.Key) {
+		panic("onCleanup: key not in region")
+	}
+	var resp kvrpcpb.CmdCleanupResponse
+	err := h.mvccStore.Cleanup(req.Key, req.GetStartVersion())
+	if err != nil {
+		if commitTS, ok := err.(ErrAlreadyCommitted); ok {
+			resp.CommitVersion = proto.Uint64(uint64(commitTS))
+		} else {
+			resp.Error = extractKeyError(err)
+		}
+	}
+	return &resp
+}
+
+func (h *rpcHandler) onCommitThenGet(req *kvrpcpb.CmdCommitThenGetRequest) *kvrpcpb.CmdCommitThenGetResponse {
+	if !h.keyInRegion(req.Key) {
+		panic("onCommitThenGet: key not in region")
+	}
+	val, err := h.mvccStore.CommitThenGet(req.Key, req.GetLockVersion(), req.GetCommitVersion(), req.GetGetVersion())
+	if err != nil {
+		return &kvrpcpb.CmdCommitThenGetResponse{
+			Error: extractKeyError(err),
+		}
+	}
+	return &kvrpcpb.CmdCommitThenGetResponse{
+		Value: val,
+	}
+}
+
+func (h *rpcHandler) onRollbackThenGet(req *kvrpcpb.CmdRollbackThenGetRequest) *kvrpcpb.CmdRollbackThenGetResponse {
+	if !h.keyInRegion(req.Key) {
+		panic("onRollbackThenGet: key not in region")
+	}
+	val, err := h.mvccStore.RollbackThenGet(req.Key, req.GetLockVersion())
+	if err != nil {
+		return &kvrpcpb.CmdRollbackThenGetResponse{
+			Error: extractKeyError(err),
+		}
+	}
+	return &kvrpcpb.CmdRollbackThenGetResponse{
+		Value: val,
+	}
+}
+
+func (h *rpcHandler) onBatchGet(req *kvrpcpb.CmdBatchGetRequest) *kvrpcpb.CmdBatchGetResponse {
+	for _, k := range req.Keys {
+		if !h.keyInRegion(k) {
+			panic("onBatchGet: key not in region")
+		}
+	}
+	pairs := h.mvccStore.BatchGet(req.Keys, req.GetVersion())
+	return &kvrpcpb.CmdBatchGetResponse{
+		Pairs: extractPairs(pairs),
+	}
+}
+
+func extractKeyError(err error) *kvrpcpb.KeyError {
+	if locked, ok := err.(*ErrLocked); ok {
+		return &kvrpcpb.KeyError{
+			Locked: &kvrpcpb.LockInfo{
+				Key:         locked.Key,
+				PrimaryLock: locked.Primary,
+				LockVersion: proto.Uint64(locked.StartTS),
+			},
+		}
+	}
+	if retryable, ok := err.(ErrRetryable); ok {
+		return &kvrpcpb.KeyError{
+			Retryable: proto.String(retryable.Error()),
+		}
+	}
+	return &kvrpcpb.KeyError{
+		Abort: proto.String(err.Error()),
+	}
+}
+
+func extractKeyErrors(errs []error) []*kvrpcpb.KeyError {
+	var errors []*kvrpcpb.KeyError
+	for _, err := range errs {
+		errors = append(errors, extractKeyError(err))
+	}
+	return errors
+}
+
+func extractPairs(pairs []Pair) []*kvrpcpb.KvPair {
```

Comment:
The name `extractPairs` doesn't properly decribe the work it does, it's more like convert pairs to `*kvrpcpb.KvPair`.